### PR TITLE
Maxs branch

### PR DESCRIPTION
**Pull request by mgisb003**

### CODE REVIEW

**Comment by mgisb003** [comment]
added topic tab to navbar and pulled up different tab names.

**Review by AdamDKing — COMMENTED** [review]
github 500 check

File: src/app/models/User.ts [hunk 1]
```diff
@@ -8,4 +8,6 @@ export class User {
     expertTags: Tag[];
     questions: Question[];
     responses: Response[];
-}
\ No newline at end of file
+}
+
+
```
Comment (AdamDKing) [inline src/app/models/User.ts:16]: avoid making changes like this

File: src/app/models/topic.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export class Topic {
+}
```
Comment (AdamDKing) [inline src/app/models/topic.ts]: We're mostly using interfaces elsewhere, does this need to be a class?

File: src/app/services/tags.service.ts [hunk 3]
```diff
@@ -14,10 +14,27 @@ const httpOptions = {
 export class TagService {
   constructor(private http: HttpClient) {}
 
+  tagId:number;
+  tagname:string;
+
```
Comment (AdamDKing) [inline src/app/services/tags.service.ts]: remove this state from the service unless absolutely necessary

File: src/app/services/users.service.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class UsersService {
+  constructor(private http: HttpClient) { }
+  tagId:number;
+
+  getExpertTags(): Observable<any[]>{
+    return this.http.get<any[]>(`${environment.userUri}`);
+  }
```
Comment (AdamDKing) [inline src/app/services/users.service.ts]: remove state, and why is this in a UsersService?

File: src/app/static/choose-topic-page/topic-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-topic-page',
+  templateUrl: './topic-page.component.html',
+  styleUrls: ['./topic-page.component.css']
+})
+export class TopicPageComponent implements OnInit {
+
+  tag: Tag[];
```
Comment (AdamDKing) [inline src/app/static/choose-topic-page/topic-page.component.ts]: change this to tags if its an array of Tag objects

File: src/app/static/choose-topic-page/topic-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  tagName: string;
```
Comment (AdamDKing) [inline src/app/static/choose-topic-page/topic-page.component.ts]: What is this for?

File: src/app/static/choose-topic-page/topic-page.component.html [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
+<div class="container">
+        <button class="btn text-light col-md-2" *ngFor="let tag of tag" (click)="viewTopic(tag.id)">
```
Comment (AdamDKing) [inline src/app/static/choose-topic-page/topic-page.component.html]: "let tag of tags"

File: src/app/static/topic-experts/topic-experts.component.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      this.experts = expert;
+      this.filterExperts();
+    });
+  }
+
+  filterExperts(): void {
+    for (let i = 0; i < this.experts.length; i++) {
+      this.expert = this.experts[i];
+      for (let j = 0; j < this.expert.expertTags.length; j++) {
+        if (this.expert.expertTags[j].id === this.getTopicId) {
```
Comment (AdamDKing) [inline src/app/static/topic-experts/topic-experts.component.ts]: This logic should be cleaned up.  Don't use fields when local variables will work, and make sure the variable names actually reflect what values are stored in them.

**Review by AdamDKing — CHANGES_REQUESTED** [review]
Lets remove state from the services and remove (most/all) HTTP calls from components, instead making calls in the specific-tag-info component and then passing those objects into subcomponents.  Let's also remove components that aren't done to but down on the size of this PR, so no webinars/AMA.   I'll have more feedback once these comments are resolved.  I'm happy to sit and chat about suggested changes.

File: src/app/models/topic.ts (hunk 2 above)
Comment (AdamDKing) [reply src/app/models/topic.ts]: It should also probably have fields

File: src/app/services/tags.service.ts (hunk 3 above)
Comment (mgisb003) [reply src/app/services/tags.service.ts]: tagname state removed, tagId necessary

File: src/app/models/topic.ts (hunk 2 above)
Comment (mgisb003) [reply src/app/models/topic.ts]: deleted class, wasn't necessary

File: src/app/static/choose-topic-page/topic-page.component.ts (hunk 5 above)
Comment (mgisb003) [reply src/app/static/choose-topic-page/topic-page.component.ts]: done

File: src/app/static/choose-topic-page/topic-page.component.html (hunk 7 above)
Comment (mgisb003) [reply src/app/static/choose-topic-page/topic-page.component.html]: done

File: src/app/static/choose-topic-page/topic-page.component.ts (hunk 6 above)
Comment (mgisb003) [reply src/app/static/choose-topic-page/topic-page.component.ts]: tagName deleted tag converted to tags for an array of tags

File: src/app/models/User.ts (hunk 1 above)
Comment (mgisb003) [reply src/app/models/User.ts:16]: whoops

File: src/app/services/users.service.ts (hunk 4 above)
Comment (mgisb003) [reply src/app/services/users.service.ts]: changed to getExperts

File: src/app/static/topic-experts/topic-experts.component.ts (hunk 8 above)
Comment (mgisb003) [reply src/app/static/topic-experts/topic-experts.component.ts]: done

**Pull request closed without merging**